crates/ra_lsp_server/src/markdown.rs

```diff
@@ -1,7 +1,11 @@
-pub(crate) fn mark_fenced_blocks_as_rust(src: &str) -> String {
+pub(crate) fn format_docs(src: &str) -> String {
     let mut processed_lines = Vec::new();
     let mut in_code_block = false;
     for line in src.lines() {
+        if in_code_block && line.trim_start().starts_with("# ") {
```

Comment:
Hmm, this is interesting. It looks like lines need to start with `# ` (the space is required), but lines with only a `#` are also hidden.

Reply:
Hm, I am not sure about this.
If rust section of docs contains `#[derive(Foo)]`, it is not ignored by rustdoc (I might be wrong here)

Reply:
I meant this case:

```rust
/// ```
/// # use std::collections::HashMap;
/// #
/// // the line above doesn't show up in rustdoc, even if it doesn't start with "# "
/// fn foo() { }
/// ```
```

Reply:
```rust
if in_code_block {
    let trimmed = line.trim();
    if trimmed.starts_with("# ") || trimmed == "#" {
        continue;
    }
```

Reply:
Oh, of course you're right, i missed this case.

Reply:
There's also the case of `/// #     `, of course (spaces at the end).

Reply:
I'v updated PR. The only this I am not sure about is tabs. Can they appear at this stage?

Reply:
No idea. The interesting case is probably `"/// #\tx"`.

Reply:
Fixed